sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosAsyncContainer.java

```diff
@@ -1310,10 +1345,12 @@ public <TContext> Flux<CosmosBulkOperationResponse<TContext>> executeBulkOperati
             bulkOptions = new CosmosBulkExecutionOptions();
         }
 
-        final CosmosBulkExecutionOptions cosmosBulkExecutionOptions = bulkOptions;
+        CosmosBulkExecutionOptions clonedOptions = bulkExecutionOptionsAccessor.clone(bulkOptions);
+        CosmosBulkExecutionOptionsImpl requestOptionsInternal = bulkExecutionOptionsAccessor.getImpl(clonedOptions);
+        applyPolicies(OperationType.Batch, ResourceType.Document, requestOptionsInternal, this.bulkSpanName);
 
         return Flux.deferContextual(context -> {
-            final BulkExecutor<TContext> executor = new BulkExecutor<>(this, operations, cosmosBulkExecutionOptions);
+            final BulkExecutor<TContext> executor = new BulkExecutor<>(this, operations, clonedOptions);
```

Comment:
should we use `requestOptionsInternal` here instead?

Reply:
Refactored so that bulkexecutor use CosmosBulkExecutionOptionsImpl instead of CosmosBulkExecutionOptions